src/components/device/ha-device-action-picker.ts

```diff
@@ -0,0 +1,33 @@
+import { customElement } from "lit-element";
+import {
+  DeviceAction,
+  fetchDeviceActions,
+  localizeDeviceAutomationAction,
+} from "../../data/device_automation";
+import "../../components/ha-paper-dropdown-menu";
+import { HaDeviceAutomationPicker } from "./ha-device-automation-picker";
+
+@customElement("ha-device-action-picker")
+class HaDeviceActionPicker extends HaDeviceAutomationPicker<DeviceAction> {
+  protected NO_AUTOMATION_TEXT = "No actions";
+  protected UNKNOWN_AUTOMATION_TEXT = "Unknown action";
+
+  constructor() {
+    super(
+      localizeDeviceAutomationAction,
+      fetchDeviceActions,
+      (deviceId?: string) => ({
+        device_id: deviceId || "",
```

Comment:
Why is there both a `device_id` and `device` ?

Reply:
For conditions, we specify the type of condition like this:
`condition: state` or `condition: device`
Likewise, for triggers we can say:
`platform: sun` or `platform: device`

For actions, there's nothing like that we just look for keys in the configuration dict:
https://github.com/home-assistant/home-assistant/blob/33e1b44b3a5724e254a4d60691c881ef3d687a59/homeassistant/helpers/script.py#L55-L72

I didn't dare to add a check for `device_id` for fear of breaking the default case so went with `device` instead.

Currently, the device actions have this format:
```
{
device: null
device_id: "1ff6a018e8a84e69807063ca84d566d3"
domain: "light"
entity_id: "light.led_trappan_nere"
type: "turn_off"
}
```

Maybe it can be simplified like this:
```
{
device: "1ff6a018e8a84e69807063ca84d566d3"
domain: "light"
entity_id: "light.led_trappan_nere"
type: "turn_off"
}
```
or:
```
{
device_id: "1ff6a018e8a84e69807063ca84d566d3"
domain: "light"
entity_id: "light.led_trappan_nere"
type: "turn_off"
}
```

Reply:
So I am confused, that if we define `condition: device`, why would we still need a `device` and `device_id` key?

Reply:
Nevermind, modified such that only `device_id` is needed for device action, also needs backend change: home-assistant/home-assistant#26536